src/main/java/Duke.java

```diff
@@ -1,10 +1,116 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static Task[] items = new Task[100];
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        greet();
+        boolean shouldExit = false;
+        Scanner in = new Scanner(System.in);
+        while(!shouldExit) {
+            String userInput = in.nextLine();
+            String[] userInputArray = (userInput.split(" "));
+            // first item in userInputArray is the command. e.g. "add" will be stored in command from "add this task"
+            String command = userInputArray[0];
+            if (command.equals("bye")) {
+                shouldExit = true;
+                separator();
+                break;
+            }
+
+            if (command.equals("list")) {
+                list();
+            } else if (command.equals("done")) {
+                int taskId = Integer.parseInt(userInputArray[1]);
+                markAsDone(taskId);
+            } else if (command.equals("todo")) {
+                String todo = userInput.replace("todo", "").trim();
+                addTodo(todo);
+            } else if (command.equals("deadline")) {
+                int deadlineDateIndex = userInput.indexOf("/by");
+                String title = userInput.substring(0, deadlineDateIndex).replace("deadline", "");
+                String deadlineDate = userInput.substring(deadlineDateIndex+3, userInput.length());
+                addDeadline(title.trim(), deadlineDate.trim());
+            } else if (command.equals("event")) {
+                int atIndex = userInput.indexOf("/at");
+                String title = userInput.substring(0, atIndex).replace("event", "");
+                String atDate = userInput.substring(atIndex+3, userInput.length());
```

Comment:
Consider adding spaces between the + operator here, which seems to be inconsistent with the rest of the code.